props/saber_fett263_buttons.h

```diff
@@ -293,248 +659,3857 @@ class SaberFett263Buttons : public virtual PropBase {
+    MENU_IGNITION_OPTION,
+    MENU_IGNITION_TIME,
+    MENU_RETRACTION_OPTION,
+    MENU_RETRACTION_TIME,
+    MENU_CLASH_LOCATION,
+    MENU_DRAG_SIZE,
+    MENU_EMITTER_SIZE,
+    MENU_PREON_OPTION,
+    MENU_PREON_SIZE,
+    MENU_DIM_BLADE,
+    MENU_CLASH_THRESHOLD,
+    MENU_SWINGON_SPEED,
+    MENU_FORCEPUSH_LENGTH,
+    MENU_LOCKUP_DELAY,
+    MENU_CLASH_DETECT,
+    MENU_MAX_CLASH,
+    COLOR_BASE,
+    COLOR_ALT,
+    COLOR_BLAST,
+    COLOR_CLASH,
+    COLOR_LOCKUP,
+    COLOR_DRAG,
+    COLOR_LB,
+    COLOR_STAB,
+    COLOR_PREON,
+    COLOR_IGNITE,
+    COLOR_RETRACT,
+    COLOR_PSTOFF,
+    COLOR_SWING,
+    COLOR_EMITTER,
+    COLOR_OFF,
+#endif  
+  };
+
+// Edit Mode Menu Select (PWR Button)
+  void MenuChoice() {
+    const char* tmp; 
+    switch (menu_type_) {
+      case MENU_PRESET:
+        PlayMenuSound("mselect.wav");
+        menu_type_ = MENU_TOP;
+        menu_ = false;
+        break;
+      case MENU_REHEARSE:
+        PlayMenuSound("rehrsbgn.wav");
+        rehearse_ = true;
+        MenuExit();
+        FastOn();
+      case MENU_VOLUME:
+        if (SaberBase::IsOn()) {
+          menu_type_ = MENU_SETTING_SUB;
+          MenuSave();
+        } else {
+          MenuExit();
+        }
+        break;  
+#ifdef FETT263_EDIT_MODE_MENU        
+      case MENU_TOP:
+        switch (menu_top_pos_) {
+          case 1:   
+  #if NUM_BLADES == 1
+            menu_type_ = MENU_STYLE;
+            num_presets_ = current_config->num_presets;
+            blade_num_ = 1;
+            tmp = current_preset_.GetStyle(blade_num_);
+            style_num_ = FirstWord(tmp, "builtin") ? atoi(SkipWord(tmp)) : 0; 
+            style_revert_ = style_num_;               
+            PlayMenuSound("mselect.wav");
+            break;
+  #else
+            menu_type_ = MENU_BLADE_STYLE;
+            blade_num_ = 0;
+            PlayMenuSound("mblade.wav");
+            break;
+  #endif
+          case 2:
+            if (style_parser.MaxUsedArgument(current_preset_.GetStyle(1)) == 0) {
+              menu_type_ = MENU_COLOR;
+              PlayMenuSound("mselect.wav");
+              color_revert_ = SaberBase::GetCurrentVariation();
+              ToggleColorChangeMode();
+              break;
+            } else {
+  #if NUM_BLADES == 1
+            blade_num_ = 1;
+            menu_type_ = MENU_COLOR_SUB;
+            menu_sub_pos_ = 0;
+            PlayMenuSound("mcolorsb.wav");
+            effect_num_ = 0;
+            break;
+  #else
+            menu_type_ = MENU_BLADE_COLOR;
+            blade_num_ = 0;
+            PlayMenuSound("mblade.wav");
+            break;                
+  #endif
+            }
+            break; 
+          case 3:
+            menu_type_ = MENU_FONT;
+            font_num_ = 0;                                               
+            num_fonts_ = RunCommandAndGetSingleLine("list_fonts", nullptr, 0, 0, 0);
+            if (!restore_point.get()) restore_point = std::move(current_preset_.font);
```

Comment:
I think it makes more sense to set the restore point when IF we alter the current preset.
As it is, you use std::move() to set the current preset which leaves the current preset in an invalid state. (no font)
If someone were to access the bluetooth menu at the same time, things will break.

Reply:
attempted to move to MenuDial() for font and track but if you rotate more than once it is updating restore_point with last font or track, not the original.  I believe we ran into this in initial testing which is why we have it being set only once when we enter the mode.  The purpose of the restore_point is to save the "original" font or track prior to rotating through options, when it's in the dial it overwrites with each rotation thereby negating it's intent.

Reply:
That shouldn't happen if you check if the restore point is already set before setting it. (Which you do on this line.)

Reply:
Figured it out, needed to add restore_point = nullptr after using restore values, was causing restore not to update if you used more than once in a session.  Updated